Add generate unique references function AB#90092

RobOatesHistoricEngland opened the pull request:
[AB#90092](https://hedev.visualstudio.com/76e71f30-b1ad-438a-a096-89ac98712f4e/_workitems/edit/90092)

- Function has been refactored to only use database sequence to derive the primary reference number
- Refactored to work with Arches 7.6
- Include database migration to register the function
- README.md created

RobOatesHistoricEngland committed:
Add test_settings_for_docker.py to test using Docker. Add comment on how to run tests

RobOatesHistoricEngland committed:
Add generate_unique_reference_function

RobOatesHistoricEngland committed:
Tests for TestGenerateUniqueReferencesFunction

RobOatesHistoricEngland committed:
Add migration and readme

khodgkinson-he commented on tests/test_settings_for_docker.py:
```diff
@@ -0,0 +1,34 @@
+
+import os
+
+# import from the test_settings.py files in the swame folder first
```
```suggestion
# import from the test_settings.py files in the same folder first
```

khodgkinson-he reviewed (approved):
LGTM.. ^_^

RobOatesHistoricEngland committed:
Typo

aj-he commented:
I've removed @gouthamrandhi as the test plans are applied at the application integration point e.g. mariner-proj integration. 

@RobOatesHistoricEngland - can you please fix the failing checks please. looks like black formating

RobOatesHistoricEngland committed:
Clarifies setup steps for sequence handling in README

Adds guidance on setting the correct initial seed value for the
PostgreSQL sequence when installing the function in existing
Arches instances to avoid conflicts. Notes that the function
does not retroactively generate reference numbers for pre-existing
resources, ensuring users understand its limitations.

RobOatesHistoricEngland committed:
Black formatter

aj-he committed:
moves migrations dir to the app folder so it is picked up AB#90092

aj-he committed:
Filters the simpleuid and globaluid drop downs to only show supported datatype nodes AB#90092

aj-he committed:
Refactor sequence creation and retrieval logic to handle exceptions and ensure sequence existence before use AB#90092

aj-he committed:
Remove print statements from test setup and execution for cleaner output
- use `-v 3` to get verbose test info
- also ensured that setup is called in the superclass

RobOatesHistoricEngland committed:
Remove redundant enumeration

RobOatesHistoricEngland committed:
File rename

RobOatesHistoricEngland committed:
Merge branch 'release/1.0.0' into 90092_ro_add_generate_unique_references_function

aj-he commented:
The coverage check will fail as this PR adds the first set of code and tests so it will drop from 100% to some other coverage.

RobOatesHistoricEngland committed:
Remove singleton. Generate sequence based on the max of 1, initial seed, or highest found in database

RobOatesHistoricEngland committed:
Cruft

RobOatesHistoricEngland committed:
Set black formatter

RobOatesHistoricEngland committed:
Remove black line length setting and default .py file to use Black default 88 line setting

khodgkinson-he reviewed (approved):
lgtm

RobOatesHistoricEngland committed:
Update README.md

aj-he reviewed (approved):
This looks ok to me for 1.0.0
